From d6cf2a97b720b4fbc199e2a7ebc2160ccd671bcc Mon Sep 17 00:00:00 2001
From: Vijay kumar Tumati <vtumati@codeaurora.org>
Date: Tue, 24 Sep 2013 14:59:54 +0530
Subject: [PATCH 019/176] Camera: Dont handle HDR meta data while capture is in
 process

While snapshot is going on, if we handle
HDR meta data, and if HDR state is changed,
consistency problems are seen.

Change-Id: I552d5fbb3dfe29fc59961a6aa8af4677db085c09
---
 QCamera2/HAL/QCamera2HWICallbacks.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/QCamera2/HAL/QCamera2HWICallbacks.cpp b/QCamera2/HAL/QCamera2HWICallbacks.cpp
index 98a8341..a686feb 100644
--- a/QCamera2/HAL/QCamera2HWICallbacks.cpp
+++ b/QCamera2/HAL/QCamera2HWICallbacks.cpp
@@ -1178,7 +1178,8 @@ void QCamera2HardwareInterface::metadata_stream_cb_routine(mm_camera_super_buf_t
           pMetaData->is_hdr_scene_data_valid,
           pMetaData->hdr_scene_data.is_hdr_scene,
           pMetaData->hdr_scene_data.hdr_confidence);
-    if (pMetaData->is_hdr_scene_data_valid) {
+    //Handle this HDR meta data only if capture is not in process
+    if (pMetaData->is_hdr_scene_data_valid && !pme->m_stateMachine.isCaptureRunning()) {
         int32_t rc = pme->processHDRData(pMetaData->hdr_scene_data);
         if (rc != NO_ERROR) {
             ALOGE("%s: processHDRData failed", __func__);
-- 
1.8.3.1

